packages/rich-text/src/to-dom.js

```diff
@@ -287,7 +291,10 @@ export function applySelection( { startPath, endPath }, current ) {
 	// This function is not intended to cause a shift in focus. Since the above
 	// selection manipulations may shift focus, ensure that focus is restored to
 	// its previous state.
-	if ( activeElement !== ownerDocument.activeElement ) {
+	// Do not take focus if user is not in iFrame editing canvas.
+	//
+	// See: https://github.com/WordPress/gutenberg/issues/61315
+	if ( isInsideFrame && activeElement !== ownerDocument.activeElement ) {
```

Comment:
We need the [`activeElement.focus()` on line 304](https://github.com/WordPress/gutenberg/pull/61374/files#diff-e740a7cf2da6241b85a670813a4e5e8992db14b5ec1bd239a07e0a63198b5f09R304) to run for Firefox to move focus after a selection change.

When that change comes from the toolbar (indent/outdent), we want the cursor to get applied to the list item and take focus. When a change comes from the sidebar, we don't want to move focus to the canvas (like a navigation label text change). I can't find a way to differentiate between these two situations. Fixing one breaks the other.

@ellatrix, do you have any insights?

Reply:
I'm failing to understand what problem this modification to rich text solves. I remove this change, tried to edit the text in the inspector for a navigation link, and focus seems to remain there. How do I reproduce the problem without this change to rich text?

These lines in rich text are only meant to maintain focus _only_ if the active element right before changing changing the range is different after changing the range.

Reply:
### On Trunk in Chrome:
- In Chrome
- Go to a navigation item
- Edit it in the canvas (add some letters)
- Tab to the sidebar
- Continue tabbing and you'll end up in a focus loop

https://github.com/WordPress/gutenberg/assets/967608/8f02d678-e01f-4e94-82f0-1c29172e9295

### On this PR in Firefox
- In Firefox
- Add a list block
- Type something
- Add another list item
- Move to the list item toolbar
- Indent the list item using the Indent button
- Focus is lost. The active focus should be moved to the list item for typing.


https://github.com/WordPress/gutenberg/assets/967608/96143e00-c4f7-4e58-81cd-4984462396df



### This PR without the `isInsideFrame` check in Firefox

- In Firefox
- Go to a navigation item
- Edit it in the canvas (add some letters)
- Go to the block settings sidebar
- Type in the label field
- Focus will get stolen back to the canvas after one keypress

https://github.com/WordPress/gutenberg/assets/967608/1160d0a1-8619-4645-8ba7-d07d74ca9992

Reply:
@ellatrix Did Jerry's response answer your queries?